src/containers/pages/OrganizationViews/AssignPractitioners/index.tsx

```diff
@@ -0,0 +1,279 @@
+   * in a single call to Practitioners endpoint
+   * @param {typeof OpenSRPService} service - the OpenSRPService
+   */
+  const loadAllPractitioners = async (service: typeof OpenSRPService = OpenSRPService) => {
+    const serve = new service(OPENSRP_PRACTITIONER_ENDPOINT);
+    const allPractitioners = await serve.list();
+    return formatOptions(allPractitioners);
+  };
+
+  // TODO - Hack in this: typings for the changeHandler function.
+  /** This sets the state selectedOptions
+   * @param {ValueType<SelectOption>} -  the so far selected options
+   * @param {ActionMeta} - information on the change event; custom react-select event
+   */
+  const changeHandler = (chosenOptions: any, { action, removedValue, option }: any) => {
+    if (!chosenOptions) {
+      return;
+    }
+    switch (action) {
+      case 'remove-value':
+      case 'pop-value':
+        if (removedValue.isFixed) {
+          return;
+        } else {
+          const remainingOptions = selectedOptions.filter(o => o.value !== removedValue.value);
+          setSelectedOptions(remainingOptions);
+        }
+        break;
+      case 'select-option':
+        setSelectedOptions([...selectedOptions, option]);
+        break;
+      case 'clear':
+        chosenOptions = chosenOptions.filter((v: any) => !v.isFixed);
+    }
+  };
+
+  // TODO - This will initiate an api request for the same exact data each time someone types
+  /** merges the practitioner records to be shown in the dropdown and returns them as  a promise
+   * @param {string} typedChars - value of select if user types in on it
+   */
+  const promiseOptions = async (typedChars: string) => {
+    // merging practitioners that are assigned to the organization and all practitioners
+    const orgPractitioners = formatOptions(assignedPractitioners, true);
+    const allPractitioners = await loadAllPractitioners();
+    const mergedOptions = {
+      ...keyBy(orgPractitioners, option => option.value),
+      ...keyBy(allPractitioners, option => option.value),
+    };
+    return filterOptions(typedChars, values(mergedOptions));
+  };
+
+  /** handles clicking on the add button */
+  const addHandler = () => {
+    // selected options
+    const practitionerIds = selectedOptions.map(option => option.value);
+    const jsonArrayPayload = practitionerIds.map((practitionerId, index) => ({
+      active: true,
+      code: PRACTITIONER_CODE,
+      identifier: generateNameSpacedUUID(`${Date.now()} ${index}`, PRACTITIONER_ROLE_NAMESPACE),
+      organization: organization.identifier,
+      practitioner: practitionerId,
+    }));
+    const serve = new serviceClass(`${OPENSRP_PRACTITIONER_ROLE_ENDPOINT}/add`);
+    serve.create(jsonArrayPayload).then(() => {
+      loadOrgPractitioners(organization.identifier, serviceClass, fetchPractitionerRolesCreator);
+      // TODO - possible candidate for setting state on unmounted component
+      setSelectedOptions([]);
+    });
```

Comment:
Should we add a `.catch` here, even if it's empty?

Reply:
just worried that this is an error that should be avoided instead of handled.

Reply:
@p-netm but leaving it this way will result in an exception being raised - now?  Much better to somehow get a nice error message to the user than to have exceptions happen - now?  (I think we have an issue to deal with errors somewhere)

Reply:
will add the error boundary